### PR TITLE
chore: added caching example

### PR DESCRIPTION
**Pull request by aarontravass**
Added a caching example which uses cache-manager for the backend.

### CODE REVIEW

File: examples/caching/weather.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import axios from 'axios'
```
Comment (talentlessguy) [inline examples/caching/weather.js:1]: there is no need to introduce an extra dependency. you can use `https` module:

```js
import https from 'node:https'

https
  .get(`https://reqres.in/api/users`, resp => {
    let data = "";

    // A chunk of data has been recieved.
    resp.on("data", chunk => {
      data += chunk;
    });

    // The whole response has been received. Print out the result.
    resp.on("end", () => {
      let url = JSON.parse(data).message;
      console.log(url);      
    });
  })
  .on("error", err => {
    console.log("Error: " + err.message);
  })
```

**Review by talentlessguy — CHANGES_REQUESTED** [review]

File: examples/caching/weather.js (hunk 1 above)
Comment (aldy505) [reply examples/caching/weather.js:1]: Or just use global `fetch` https://nodejs.org/api/globals.html#fetch

Comment (aarontravass) [reply examples/caching/weather.js]: For some reason fetch was not available. Probably because my node version was old.

**Comment by aarontravass** [comment]
> is it possible to replace it with lru-send? cuz it's also a caching solution

let me give it a try

**Comment by aarontravass** [comment]
> is it possible to replace it with lru-send? cuz it's also a caching solution

done

File: examples/caching/weather.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+const URL = 'https://goweather.herokuapp.com/weather/'
```
Comment (talentlessguy) [inline examples/caching/weather.js:1]: Can you please move this to the file with the rest of the code plz? It's ~25 LOC anyway

**Review by talentlessguy — CHANGES_REQUESTED** [review]

**Review by aldy505 — APPROVED** [review]

**Pull request merged**